mesop/runtime/context.py

```diff
@@ -120,9 +125,36 @@ def serialize_state(self) -> pb.States:
       states.states.append(pb.State(data=serialize_dataclass(state)))
     return states
 
+  def diff_state(self) -> pb.States:
+    states = pb.States()
+    for (key, state), previous_state in zip(
+      self._states.items(), self._previous_states.values()
+    ):
+      states.states.append(pb.State(data=diff_state(previous_state, state)))
+      # If state has not changed no need to update the previous state since it should be
+      # the same.
+      #
+      # One thing to note is that there seems to be an issue multiple states with the
+      # same key can exist (i.e. usage in Mesop demo app). It's unclear if this causes
+      # problem here.
+      if state != self._previous_states[key]:
+        self._previous_states[key] = copy.deepcopy(state)
```

Comment:
I don't understand why we need to update _previous_states because, IIUC, diff_state is called at most once per context lifetime.

Reply:
Great catch. I had this in before I added the change to only update only send the state back once.

Reply:
Not sure if you meant to change this, but I think we can remove any mutations to previous_states - see other comment.

Reply:
With #394 - do we still need to mutate previous_states here?

Reply:
Yeah, even with #394 this wasn't needed the update to only send state at the end of the stream. It should be removed already. I think I just forgot to add a follow up comment here.